app/src/main/java/com/iic/shopingo/ui/request_flow/activities/SelectShopperActivity.java

```diff
@@ -10,23 +11,42 @@
 import butterknife.InjectView;
 import butterknife.OnItemClick;
 import com.iic.shopingo.R;
+import com.iic.shopingo.services.CurrentLocationProvider;
 import com.iic.shopingo.ui.request_flow.views.SelectShopperListItemView;
 import java.util.ArrayList;
 import java.util.List;
 
 public class SelectShopperActivity extends ActionBarActivity {
+  private final long REQUEST_INTERVAL = 10 * 1000; // 10 seconds in millisecnods
+
   @InjectView(R.id.select_shopper_list)
   ListView shopperList;
 
   private SelectShopperAdapter adapter;
 
+  private CurrentLocationProvider locationProvider;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_select_shopper);
     ButterKnife.inject(this);
 
-    adapter = new SelectShopperAdapter(new ArrayList<SelectShopperAdapter.Shopper>());
+    locationProvider = new CurrentLocationProvider(this, REQUEST_INTERVAL, new CurrentLocationProvider.LocationUpdatesListener() {
+      @Override
+      public void onLocationUpdated(Location location) {
+        adapter.setUserLocation(location);
+      }
+
+      @Override
+      public void onConnectionStop() {}
+
+      @Override
+      public void onConnectionFail() {}
+    });
```

Comment:
Not critical but in Java there is this pattern where you create an "adapter" class that implements the interface and provide empty implementations for all its methods, then when you want to use it you can simply override the methods you are interested in.
For example:

```
locationProvider = new CurrentLocationProvider(this, REQUEST_INTERVAL, new LocationUpdatesAdapter() {
@Override
public void onLocationUpdated(Location location) {
 ...
}
});
```